### PR TITLE
Add keys

### LINKED ISSUE #50: Start publishing releases. Deprecate scalaz/ioeffect
1) Now that this properly exists, scalaz/ioeffect technically no longer needs to exist, does it?
2) Can we tag a release for ZIO, so people using scalaz/ioeffect can migrate onto a stable version?

### LINKED ISSUE #172: commit hash releases badly broken
Those non-SNAPSHOT commit hash versions are considered "stable" by Sonatype, not snapshots.

Hence, these artifacts need to go through the correct release procedure.
Here is the workflow with the steps we are already doing checked off:

- [ ] Create POM with license, website, developer list
- [X] Generate artifacts
- [ ] Sign artifacts
- [X] Publish artifacts
- [ ] Release staging repository

Need to add this to the top of the `build.sbt`:

```
import xerial.sbt.Sonatype._

licenses in ThisBuild := Seq("APL2" -> url("http://www.apache.org/licenses/LICENSE-2.0.txt"))
homepage in ThisBuild := Some(url("https://scalaz.github.io/scalaz-zio/"))
developers in ThisBuild := List(
  Developer(id = "jdegoes", name = "John De Goes", url = url("http://degoes.net"), email = "john@degoes.net")
)
publishMavenStyle := true
```

The `.travis.yml` should invoke `publishSigned` and `sonatypeRelease` instead of `publish`.

A GPG signing key will be required, the password to the key should please be sent to me.

This may be very useful reading: https://alexn.org/blog/2017/08/16/automatic-releases-sbt-travis.html

### PR DESCRIPTION
**Pull request by NeQuissimus**
Closes #50 
Closes #172 

As usual, we can only tell whether the ENV variables for Travis work once the change has been merged into `master` :(

This works for me locally, but you never know...

/cc @pshirshov

### CODE REVIEW

**Comment by jdegoes** [comment]
Well, let's try it... 🤞

**Comment by jmcardon** [comment]
wait uhh... are you guys pushing private keys to github?

File: project/secring.gpg [hunk 1]
```diff
@@ -0,0 +1,60 @@
+-----BEGIN PGP PRIVATE KEY BLOCK-----
```
Comment (jmcardon) [inline project/secring.gpg]: W A T

**Comment by pshirshov** [comment]
I'm around

```
[error] (Compile / scalafmtSbtCheck) /home/travis/build/scalaz/scalaz-zio/build.sbt isn't formatted properly!
```

:D

**Comment by pshirshov** [comment]
Ah, yeah, check that private key is actually encrypted. Though it should be.

**Comment by NeQuissimus** [comment]
Formatted, private key can't be used without passphrase.

**Comment by pshirshov** [comment]
Green, you may merge

**Pull request merged**